pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -2331,7 +2367,282 @@ def ReuseObjectiveFunctionRule(model):
+        base_dir = Path(this_file_dir())
+        if model.config.desalination_model == DesalinationModel.mvc:
+            keras_surrogate = KerasSurrogate.load_from_folder(
+                str(base_dir / "mvc_keras")
+            )
+        elif model.config.desalination_model == DesalinationModel.md:
+            keras_surrogate = KerasSurrogate.load_from_folder(
+                str(base_dir / "md_keras")
+            )
+
+        for i in model.s_R:
+            for t in model.s_T:
+                if model.p_chi_DesalinationSites[i]:
+                    # Build the model with non-zero outputs
+                    cap = model.v_T_Treatment_scaled[i, t]
+                    model.surrogate_costs[i, t].build_model(
+                        keras_surrogate,
+                        formulation=KerasSurrogate.Formulation.RELU_BIGM,
+                        input_vars=[model.inlet_salinity[i], model.recovery[i], cap],
+                        output_vars=[
+                            model.v_C_TreatmentCapEx_site_time[i, t],
+                            model.v_C_Treatment_site[i, t],
+                        ],
+                    )
+                else:
+                    # If not a desalination site, fix the outputs to zero
+                    model.v_T_Treatment_scaled[i, t].fix(0)
+                    model.v_C_TreatmentCapEx_site_time[i, t].fix(0)
+                    model.v_C_Treatment_site[i, t].fix(0)
+
+        def flowBinRule(model, r, t):
+            return model.v_T_Treatment_scaled[r, t] >= 0 + 1e-6 - model.BigM * (
+                1 - model.vb_y_flow_ReLU[r, t]
+            )
+
+        model.flowBin = Constraint(
+            model.s_R,
+            model.s_T,
+            rule=flowBinRule,
+            doc="Flow binary to set to 0 if flow is 0, else 1",
+        )
+
+        def flowMaxRule(model, r, t):
+            return (
+                model.v_T_Treatment_scaled[r, t]
+                <= model.BigM * model.vb_y_flow_ReLU[r, t]
+            )
+
+        model.flowMax = Constraint(
+            model.s_R,
+            model.s_T,
+            rule=flowMaxRule,
+            doc="Flow binary to set to 0 if flow is 0 else 1",
+        )
+
+        def OpexTreatmentRule(model, r, t):
+            return model.v_C_Treatment_site_ReLU[r, t] >= model.v_C_Treatment_site[
+                r, t
+            ] - model.BigM * (1 - model.vb_y_flow_ReLU[r, t])
+
+        model.OpexTreatment = Constraint(
+            model.s_R, model.s_T, rule=OpexTreatmentRule, doc="Opex based on binary"
+        )
+
+        def treatmentCost(model):
+            if model.decision_period == pyunits.day:
+                annual_factor = 365
+            elif model.decision_period == pyunits.week:
+                annual_factor = 52
+            elif model.decision_period == pyunits.fortnight:
+                annual_factor = 26
+            elif model.decision_period == pyunits.month:
```

Comment:
I suggest adding another `else` clause and throwing an error in case the decision period is not selected as day, week, fortnight, or month.

Reply:
I shall push this changes. I hope this does not fail the codecov patch coverage.

Reply:
@JavalVyas2000 I think it will still pass the codecov check, but we'll see. Resolving.